spec/models/computacenter/api/cap_usage_update_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.describe Computacenter::API::CapUsageUpdate do
+  let(:args) do
+    {
+      'capType' => 'DfE_RemainThresholdQty|Std_Device',
+      'shipTo' => '123456',
+      'capAmount' => 100,
+      'usedCap' => 20,
+    }
+  end
+  let(:update) { described_class.new(args) }
+
+  describe '#initialize' do
+    context 'given a hash with string keys' do
+      it 'sets cap_type to the given capType' do
+        expect(update.cap_type).to eq(args['capType'])
+      end
+
+      it 'sets ship_to to the given shipTo' do
+        expect(update.ship_to).to eq(args['shipTo'])
+      end
+
+      it 'sets cap_amount to the given capAmount' do
+        expect(update.cap_amount).to eq(args['capAmount'])
+      end
+
+      it 'sets cap_used to the given usedCap' do
+        expect(update.cap_used).to eq(args['usedCap'])
+      end
+    end
+  end
+
+  describe '#apply!' do
+    let!(:school) { create(:school, computacenter_reference: '123456') }
+    let!(:allocation) { create(:school_device_allocation, school: school, device_type: 'std_device') }
+
+    before do
+      allow(School).to receive(:find_by_computacenter_reference!).with('123456').and_return(school)
+      allow(school.device_allocations).to receive(:find_by_device_type!).with('std_device').and_return(allocation)
```

Comment:
This is more personal preference than anything else, but the stubbing makes the spec a bit harder to read and adds extra lines. If there a strong need to avoid just using the underlying ActiveRecord without stubbing (we've got hundreds of specs doing this already)?

Reply:
Just the general principle of 'if you need genuine database interactions, that should be an integration/system/etcetc test'. But I agree, overuse of stubbing leads to "testing that it does what it does", and this admittedly felt a little.  bit icky at the time of writing